### DISCUSSION TITLE
Unable to install nwg-shell on Fedora 39

### DISCUSSION THREAD

**Discussion by ekremkaraca** [nwg-piotr/nwg-shell#241] [Q&A] [answered]
Hi,

When I run the command below to install nwg-shell, I got the error below. It seems there are dependencies not available for Fedora 39. Can you make them available for it?

```sh
# dnf install nwg-shell
Last metadata expiration check: 0:05:23 ago on Tue 12 Dec 2023 01:09:21 PM +03.
Error: 
 Problem: conflicting requests
  - nothing provides nwg-drawer needed by nwg-shell-0.5.23-1.fc39.noarch from copr:copr.fedorainfracloud.org:tofik:nwg-shell
  - nothing provides nwg-menu needed by nwg-shell-0.5.23-1.fc39.noarch from copr:copr.fedorainfracloud.org:tofik:nwg-shell
(try to add '--skip-broken' to skip uninstallable packages)
```

**Comment by nwg-piotr** [comment]
Sorry, but Fedora maintainers seem to have dropped further work on nwg-shell as a whole. I haven't heard from them for months.

Personally I don't do any packaging other than for Arch Linux (AUR). You can try and get in touch with @kuba3351 or other package maintainers. In case it fails, please let me know: I'll remove Fedora from the Wiki.

**Reply by ekremkaraca** [reply]
@nwg-piotr thanks for reply. so sorry to hear it.

**Reply by kuba3351** [reply]
Hello

I am not a package maintainer, I just wrote wiki and install script based on someone else work. I does not have necessary knowledge about packaging to take it.

I successfully installed nwg-shell on Fedora 39 by using the nwg-drawer compliled for Fedora 38 but this is not a solution, just workaround.

Some dependencies are still updating because package maintainer set up a script that updates packages to latest version, so e.g recent nwg-panel is available in the repositories and compiled for Fedora 39

**Reply by nwg-piotr** [reply]
@kuba3351 

> I am not a package maintainer (...)

Yes, but I thought you probably knew what's going on. :) Thanks for replying!

**Reply by ekremkaraca** [reply]
@kuba3351 thanks for reply, maybe we should ask @t0fik who runs copr repo.

**Reply by t0fik** [reply]
Which packages are missing? I'll add them to my copr

**Reply by kuba3351** [reply]
There are no missing packages, just nwg-drawer and nwg-menu does not have a build for Fedora 39 so, it is not possible to install nwg-shell on Fedora 39 without using these packages compiled for f38

**Reply by t0fik** [reply, marked as answer]
I've missed package naming problem with `nwg-drawer` package. In an hour everything should work perfectly. Never less packages should be forked from f38 to f39 repositories and running f38 on f39 shouldn't be a problem.

**Reply by ekremkaraca** [reply]
@t0fik yep, I can now install nwg-shell properly. Thanks so much!

**Reply by twiclo** [reply]
Hate to be that guy but I can't get it to install on Fedora 40
```[twiclo@sato ~]$ dnf install nwg-shell
Error: This command has to be run with superuser privileges (under the root user on most systems).
[twiclo@sato ~]$ sudo !!
sudo dnf install nwg-shell
Copr repo for nwg-shell owned by tofik                      12 kB/s | 9.5 kB     00:00
Copr copr.fedorainfracloud.org/tofik/nwg-shell runtime dep  25 kB/s | 7.7 kB     00:00
Copr copr.fedorainfracloud.org/tofik/nwg-shell runtime dep 7.7 kB/s | 2.1 kB     00:00
Error:
 Problem: package nwg-shell-0.5.34-1.fc40.noarch from copr:copr.fedorainfracloud.org:tofik:nwg-shell requires gtklock, but none of the providers can be installed
  - conflicting requests
  - nothing provides libgtk-session-lock.so.0()(64bit) needed by gtklock-3.0.0-2.fc40.x86_64 from coprdep:copr.fedorainfracloud.org:tofik:sway```

**Reply by kuba3351** [reply]
Please activate this copr also and it should install fine: https://copr.fedorainfracloud.org/coprs/mochaa/gtk-session-lock/

**Reply by t0fik** [reply]
`gtk-session-lock` by mochaa is on the way to be part of official Fedora repositories, so this dependency might be temporary.
Link to Mochaa's pagure https://pagure.io/mochaa-rpms/gtk-session-lock
